### PR TITLE
[FUSEQE-3823] Add box connector tests

### PR DESCRIPTION
**Pull request by avano**
//test: `@box`

### CODE REVIEW

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: utilities/src/main/java/io/syndesis/qe/utils/BoxUtils.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            BoxFile.Info info = new BoxFolder(this.boxConnection, SYNDESIS_BOX_DIRECTORY_ID).uploadFile(
+                    IOUtils.toInputStream(content, "UTF-8"), name);
+            fileId = info.getID();
+        } catch (Exception ex) {
+            fail("Unable to upload file to Box: ", ex);
+        }
+    }
+
+    public int getFileCount() {
+        return ((Collection<?>)new BoxFolder(this.boxConnection, SYNDESIS_BOX_DIRECTORY_ID)).size();
```
Comment (mcada) [inline utilities/src/main/java/io/syndesis/qe/utils/BoxUtils.java]: should there be a space before new?

File: utilities/src/main/java/io/syndesis/qe/utils/BoxUtils.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        return null;
+    }
+
+    /**
+     * From org.apache.camel.component.box.internal.BoxConnectionHelper
+     * @return box access token
+     */
+    private String getAccessToken() throws Exception {
+        final String csrfToken = String.valueOf(new SecureRandom().nextLong());
+        final String url = OAUTH_URL + boxAccount.getProperty("clientId")+ "&state=" + csrfToken;
```
Comment (mcada) [inline utilities/src/main/java/io/syndesis/qe/utils/BoxUtils.java]: space before +

File: utilities/src/main/java/io/syndesis/qe/utils/BoxUtils.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        if (code == null) {
+            fail("Unable to get code from " + location);
+        }
+
+        return code;
+    }
+
+    /**
+     * From org.apache.camel.component.box.internal.BoxConnectionHelper
+     * @param page html page
```
Comment (mcada) [inline utilities/src/main/java/io/syndesis/qe/utils/BoxUtils.java]: there is usually one more empty line in between comment and stuff like `@param` arguments

File: utilities/src/main/java/io/syndesis/qe/utils/BoxUtils.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+            if (name.equals(info.getName())) {
+                return new BoxFile(this.getBoxConnection(), info.getID());
+            }
+        }
+        return null;
+    }
+
+    /**
+     * From org.apache.camel.component.box.internal.BoxConnectionHelper
+     * @return box access token
```
Comment (mcada) [inline utilities/src/main/java/io/syndesis/qe/utils/BoxUtils.java]: there is usually one more empty line in between comment and stuff like `@return` arguments, at least my IDE does that

File: utilities/src/main/java/io/syndesis/qe/utils/BoxUtils.java [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+    private Account boxAccount;
+    @Getter
+    private BoxAPIConnection boxConnection;
+    @Getter
+    private static String fileId;
+
+    @PostConstruct
+    public void initClient() {
+        log.info("Getting access token");
```
[the remark this answers is not in the document]
Comment (avano) [reply utilities/src/main/java/io/syndesis/qe/utils/BoxUtils.java]: the log looks like this:

```
 INFO  [io.syndesis.qe.utils.BoxUtils:52] Getting access token
```

but I'll add it

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
the previous fail was checkstyle only, so I'm going to merge

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request merged**